### PR TITLE
feat: Add Residual Plotting Functionality

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR initialized qc module and residual plotting functionality such as plotting the ENU components and residuals together with outliers/deletions flagging.

## Reference

Issue #149 
Issue #148 

### Residuals

res

### ENU Components

enu_comps

### CODE REVIEW

**Comment by lsetiawan** [comment]
@johnbdesanto Please let me know if the demo plots above looks okay to you. The time axis seems like there might be too much ticks... currently it defaults to `25` ticks.. maybe I need to reduce this.

**Comment by johnbdesanto** [comment]
The ticks on the x-axis are a little dense. If there's a way to reduce the ticks to the 15-20 range, I suspect that that might be easier to parse the axis labels.

I was concerned that the plot may be colorblind-unfriendly, but after checking with coblis it seemed fine.

**Comment by lsetiawan** [comment]
> The ticks on the x-axis are a little dense. If there's a way to reduce the ticks to the 15-20 range, I suspect that that might be easier to parse the axis labels.

That's what I thought. I actually made the underlying function super easy to adjust the ticks, so yes we can reduce it. I set to `15` and this is what I get:

Screenshot 2023-10-16 at 4 27 55 PM

> I was concerned that the plot may be colorblind-unfriendly, but after checking with coblis it seemed fine.

Thanks for checking on that. I haven't really thought about the colors. I will change that to make sure it's the most colorblind friendly!

Really appreciate your input @johnbdesanto 😄

**Comment by lsetiawan** [comment]
### Updated demo after `n_ticks = 15` and now using a more colorblind friendly colors

res

enu_comps

File: src/gnatss/ops/qc.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+       The ISO formatted time string
+       as 'YYYY-MM-DDThh:mm:ss.dddddd'
+    """
+    astro_time = AstroTime(sec, format=fmt)
+    return astro_time.strftime("%Y-%m-%dT%H:%M:%S.%f")
+
+
+def _compute_ticks_and_labels(
+    data: pd.DataFrame, n_ticks: int = 25, time_col: str = "time"
+) -> Tuple:
```
Comment (carlosgjs) [inline src/gnatss/ops/qc.py:48]: `Tuple[??]`?

File: src/gnatss/ops/qc.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    fig, axs = plt.subplots(figsize=figsize)
+
+    # Plot residuals
+    time_labels = [constants.TIME_J2000, constants.TIME_ISO]
+    col_names = []
+    for i, col in enumerate(residuals_df.columns):
+        if col not in time_labels:
+            col_names.append(col)
+            residuals_df.plot.scatter(
+                x="time", y=col, s=2, ax=axs, c=CB_COLORS[i + 1], zorder=i + 2
```
Comment (carlosgjs) [inline src/gnatss/ops/qc.py:111]: Any chance of going over the length of `CB_COLORS`?

**Review by carlosgjs — COMMENTED** [review]
Couple of inline comments and looks like some missing tests.

File: src/gnatss/ops/qc.py (hunk 2 above)
Comment (lsetiawan) [reply src/gnatss/ops/qc.py:112]: No, this should max out to only 6 total upstream.

**Comment by lsetiawan** [comment]
> Couple of inline comments and looks like some missing tests.

Thanks! I'm skipping tests for now. Trying to get this out for user testing atm

**Pull request merged**